### PR TITLE
Added module Share [ Browser ]

### PR DESCRIPTION
**Pull request by joaoeudes7**
I added a module of Share in Core, I think so is useful.

### CODE REVIEW

File: packages/core/useShare/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+type ShareOptions = {
+  title?: string
+  files?: File[]
+  text?: string
+  url?: string
+}
+
+const _navigator = (window.navigator as NavigatorWithShare)
```
Comment (antfu) [inline packages/core/useShare/index.ts]: Please move this line inside the function.

File: packages/core/useShare/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export async function useShare<T>(shareOpts: ShareOptions) {
```
Comment (antfu) [inline packages/core/useShare/index.ts]: Look like the `<T>` is not used anywhere. Can you remove it?

File: packages/core/useShare/index.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  if (_navigator && _navigator.share) {
+    let granted = true
+
+    if (shareOpts.files && _navigator.canShare)
+      granted = _navigator.canShare({ files: shareOpts.files })
+
+    if (granted) {
+      _navigator.share(shareOpts)
+        .then(() => console.log('Successful share'))
+        .catch(err => console.log(err.message))
```
Comment (antfu) [inline packages/core/useShare/index.ts]: Using `console.log` seems not a good practice for a util function.

Maybe you can simply return the promise and let users to handle the callbacks
```js 
return _navigator.share(shareOpts)
```

File: packages/core/useShare/index.ts (hunk 2 above)
Comment (joaoeudes7) [reply packages/core/useShare/index.ts]: I forgot that hehe thanks

**Pull request merged**